core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -220,23 +225,48 @@ public void start(WorkerNetAddress address) throws IOException {
 
     // Acquire worker Id.
     BlockMasterClient blockMasterClient = mBlockMasterClientPool.acquire();
-    try {
-      RetryUtils.retry("create worker id", () -> mWorkerId.set(blockMasterClient.getId(address)),
-          RetryUtils.defaultWorkerMasterClientRetry(ServerConfiguration
-              .getDuration(PropertyKey.WORKER_MASTER_CONNECT_RETRY_TIMEOUT)));
-    } catch (Exception e) {
-      throw new RuntimeException("Failed to create a worker id from block master: "
-          + e.getMessage());
-    } finally {
-      mBlockMasterClientPool.release(blockMasterClient);
+    File file = new File(INFOFILE);
+    boolean isNewWorker = true;
+    if (file.exists()) {
+      isNewWorker = false;
+    }
+    if (!isNewWorker) {
+      try (FileInputStream input = new FileInputStream(file)) {
+        WorkerMeta.Info infoFile = WorkerMeta.Info.parseFrom(input);
+        RetryUtils.retry("get worker id",
+            () -> mWorkerId.set(blockMasterClient.getId(address, infoFile.getWorkerId())),
+            RetryUtils.defaultWorkerMasterClientRetry(
+                ServerConfiguration.getDuration(PropertyKey.WORKER_MASTER_CONNECT_RETRY_TIMEOUT)));
+        if (mWorkerId.get() != infoFile.getWorkerId()) {
+          isNewWorker = true;
+        }
+      } catch (Exception e) {
+        throw new RuntimeException(
+            "Failed to create or get old worker id: " + e.getMessage());
+      } finally {
+        mBlockMasterClientPool.release(blockMasterClient);
+      }
+    } else {
+      file.createNewFile();
```

Comment:
CreateNewFile here is not necessary, because the new FileOutputStream(file) will create a file if it is not exist as far as I know. please verify.